receiver/k8sclusterreceiver/internal/clusterresourcequota/clusteresourcequotas.go

```diff
@@ -1,114 +0,0 @@
-// Copyright The OpenTelemetry Authors
-// SPDX-License-Identifier: Apache-2.0
-
-package clusterresourcequota // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/k8sclusterreceiver/internal/clusterresourcequota"
-
-import (
-	"strings"
-
-	agentmetricspb "github.com/census-instrumentation/opencensus-proto/gen-go/agent/metrics/v1"
-	metricspb "github.com/census-instrumentation/opencensus-proto/gen-go/metrics/v1"
-	resourcepb "github.com/census-instrumentation/opencensus-proto/gen-go/resource/v1"
-	quotav1 "github.com/openshift/api/quota/v1"
-	conventions "go.opentelemetry.io/collector/semconv/v1.6.1"
-	corev1 "k8s.io/api/core/v1"
-
-	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/k8sclusterreceiver/internal/constants"
-	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/k8sclusterreceiver/internal/utils"
-)
-
-var clusterResourceQuotaLimitMetric = &metricspb.MetricDescriptor{
-	Name:        "openshift.clusterquota.limit",
-	Description: "The configured upper limit for a particular resource.",
-	Type:        metricspb.MetricDescriptor_GAUGE_INT64,
-	LabelKeys: []*metricspb.LabelKey{{
-		Key: "resource",
-	}},
-}
-
-var clusterResourceQuotaUsedMetric = &metricspb.MetricDescriptor{
-	Name:        "openshift.clusterquota.used",
-	Description: "The usage for a particular resource with a configured limit.",
-	Type:        metricspb.MetricDescriptor_GAUGE_INT64,
-	LabelKeys: []*metricspb.LabelKey{{
-		Key: "resource",
-	}},
-}
-
-var appliedClusterResourceQuotaLimitMetric = &metricspb.MetricDescriptor{
-	Name:        "openshift.appliedclusterquota.limit",
-	Description: "The upper limit for a particular resource in a specific namespace.",
-	Type:        metricspb.MetricDescriptor_GAUGE_INT64,
-	LabelKeys: []*metricspb.LabelKey{
-		{
-			Key: "resource",
-		},
-		{
-			Key: conventions.AttributeK8SNamespaceName,
-		},
-	},
-}
-
-var appliedClusterResourceQuotaUsedMetric = &metricspb.MetricDescriptor{
-	Name:        "openshift.appliedclusterquota.used",
-	Description: "The usage for a particular resource in a specific namespace.",
-	Type:        metricspb.MetricDescriptor_GAUGE_INT64,
-	LabelKeys: []*metricspb.LabelKey{
-		{
-			Key: "resource",
-		},
-		{
-			Key: conventions.AttributeK8SNamespaceName,
-		},
```

Comment:
We need to preserve the existing format and not introduce breaking changes. The metrics are emitted with `resource` and `k8s.namespace.name` attributes. This PR drops the `resource` attribute and moves  `k8s.namespace.name` to a resource attribute.

Reply:
Thanks, I've updated to match old format. However, I've run into a different issue when updating the existing test and I'm not sure what the best path forward here is.

Previously, multiple metrics of the same name would be emitted even though they share the same resource and scope, in the case of the `openshift.appliedclusterquota.limit` and `openshift.appliedclusterquota.used`. These metrics have an extra attribute for the `k8s.namespace.name` that the quota is for, but this is an attribute of the metric, not scope/resource. From [documentation](https://opentelemetry.io/docs/specs/otel/metrics/data-model/#opentelemetry-protocol-data-model-producer-recommendations), only one metric of a given name should be emitted for a resource and scope, so the metric would just have multiple datapoints. After updating to pdata, only one metric will be emitted for a given name, and then each datapoint will have the given `k8s.namespace.name` as an attribute.

My question here is what's the best way to handle this? Do we try to match the old output even though it's wrong? Do we emit metrics properly and mark it a breaking change (if it is a breaking change)?

Reply:
Also, I believe the `resourcequota` metrics had the same issue, but the unit test didn't cover this case. We shipped it with the [updated metric output](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/23419), so I assume it's either okay to update to the proper format, or we'll need to update that.

Reply:
> Previously, multiple metrics of the same name would be emitted even though they share the same resource and scope, in the case of the openshift.appliedclusterquota.limit and openshift.appliedclusterquota.used.

Good finding. This is probably a bug of OC->OTel translation. Worth filing a ticket. We definitely don't need to recreate this wrong behavior. And mdatagen generated code won't even allow that. I wouldn't worry about applying it without marking it as a breaking change. It's a different representation (several metrics vs one metric with several datapoints), but the data is pretty much the same.

Reply:
What should the bug be filed against? I'm not sure what the actual request would be if we're saying the old way was wrong, and now we're converting to the proper way? Is the action item to go through metrics that are still in the OpenCensus format and update them so they don't create multiple metrics with the same name, resource, and scope attributes?

Reply:
Actually, I misunderstood the root cause. It's not a bug in the OC->Otel conversion but rather a bug in how we have created the OC metrics in this receiver. So we are good now. Thanks!